src/nnfusion/frontend/tensorflow_import/ops/const.cpp

```diff
@@ -124,6 +125,134 @@ namespace nnfusion
                 return true;
             }
 
+            bool ValuesFromConstNode(const tensorflow::NodeDef& node,
+                                     tensorflow::TensorShapeProto* const_tensor_shape,
+                                     DataBuffer* values)
+            {
+                NNFUSION_CHECK(node.op() == "Const");
+                auto dt = node.attr().at("dtype").type();
+                // if (dt != DataTypeToEnum<T>::value)
+                // {
+                //     std::stringstream ss;
+                //     ss << "Invalid data type defined for Const. Defined: " << dt;
+                //     return false;
+                // }
+
+                // TensorProto represents the content of the tensor in either <type>_val or
+                // tensor_content.
+                const tensorflow::TensorProto& tensor = node.attr().at("value").tensor();
+                // typename checkpoint::SaveTypeTraits<T>::RepeatedField* tensor_values =
+                //     checkpoint::MutableTensorProtoData<T>(const_cast<TensorProto*>(&tensor));
+
+                const tensorflow::TensorShapeProto& shape = tensor.tensor_shape();
+                *const_tensor_shape = shape;
+
+                const auto tensor_content_size = tensor.tensor_content().size();
+                // NNFUSION_LOG(INFO) << values->get_type() << ": tensor_size = " << tensor_content_size << ", type size = " << values->get_type().size();
+                NNFUSION_CHECK(0 == tensor_content_size % values->get_type().size());
+
+                int64_t n_elements = 1;
+                for (size_t i = 0; i < shape.dim_size(); i++)
+                {
+                    if (shape.dim(i).size() < 0)
+                    {
+                        return false;
+                        // return errors::InvalidArgument(
+                        //     "Const node has empty tensor and an unknown dimension size");
+                    }
+                    n_elements *= shape.dim(i).size();
+                }
+
+                // If tensor_content_size is zero, we'll have to take the values from
+                // int_val, float_val, etc.
+                if (tensor_content_size == 0)
+                {
+                    values->resize(n_elements);
+                    for (size_t i = 0; i < n_elements; i++)
```

Comment:
why not just std::memcpy here?

Reply:
Data from tensorflow pb can only be accessed by its special interface.

Reply:
`.data()` method returns underlying array pointer. see https://github.com/protocolbuffers/protobuf/blob/c981d39d1e3d033c4f00dd48d16e8bdce55a3861/src/google/protobuf/repeated_field.h#L242

Reply:
Thanks for the information, I will look into it.